client/views/components/nodeInput/nodeInput.js

```diff
@@ -38,15 +55,27 @@ Template.nodeInputComponent.events({
       t.$('.search').focus();
     }, 100);
   },
-  'mousedown .add': function(e, t) {
+  'mousedown .search-list .list-group-item': function(e, t) {
     var selected = t.data.selected.get();
     selected.push(this);
     t.data.selected.set(selected);
-    Template.instance().searchResult.set([]);
-    Template.instance().searchDisplay.set(false);
+    resetSearchState();
+    return false;
   },
   'mousedown .remove': function(e, t) {
     var selected = t.data.selected.get();
     t.data.selected.set(Lazy(selected).without(this).toArray());
+  },
+  'mouseover .list-group-item': function(e, t) {
+    t.$('.search-list .list-group-item').removeClass('hover');
+    t.$(e.currentTarget).addClass('hover');
+    t.highlightedResult.set(t.$(e.currentTarget).index());
   }
 });
+
+// Auxiliary functions
+var resetSearchState = function(){
```

Comment:
Added this auxiliary function to DRY up event handling code. Please comment before merging.